File: algorithms/easy/sum_of_squares_of_special_elements.ex
defmodule Solution do
  @spec sum_of_squares(nums :: [integer]) :: integer
  def sum_of_squares(nums) do
    len = Enum.count(nums)

    nums
    |> Enum.with_index()
    |> Enum.reduce(0, fn {n, i}, acc ->
      if rem(len, i + 1) == 0 do
        acc + n * n
      else
        acc
      end
    end)
  end
end

File: algorithms/easy/sum_of_squares_of_special_elements.rs
impl Solution {
    pub fn sum_of_squares(nums: Vec<i32>) -> i32 {
        let mut ans = 0;
        let len = nums.len();
        for (i, n) in nums.into_iter().enumerate() {
            if len % (i + 1) == 0 {
                ans += n * n;
            }
        }
        ans
    }
}
